api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -434,8 +445,15 @@ public Workspace saveAndCloneCohortsConceptSetsAndDataSets(Workspace from, Works
   public WorkspaceAccessLevel getWorkspaceAccessLevel(
       String workspaceNamespace, String workspaceId) {
     WorkspaceACL workspaceACL = fireCloudService.getWorkspaceAcl(workspaceNamespace, workspaceId);
+    Map<String, WorkspaceAccessEntry> workspaceAcls = workspaceACL.getAcl();
+    WorkspaceAccessEntry userAcl = workspaceAcls.get(userProvider.get().getEmail());
+    // Published / featured workspaces don't give ACLs to everyone per se, they give READER access
+    // to the registered user domain email, so we check for that one if the user doesn't have access
+    // themself.
+    WorkspaceAccessEntry actualAcl =
+        userAcl == null ? workspaceAcls.get(getRegisteredUserDomainEmail()) : userAcl;
```

Comment:
if a user's ACL is NO ACCESS, should they still get READER access through the registered domain?

Reply:
If it's NO_ACCESS, then `userAcl` wouldn't be null.